frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js

```diff
@@ -552,31 +571,148 @@ export default function WaveformGraph({
+        const id = d3.select(this).attr("id");
 
-        const pvCopy = peakValleyWindows;
-        if (id === "peakLine") {
-          pvCopy[currentWell].minPeaks = y.invert(yPosition);
-          setPeakValleyWindows({ ...pvCopy });
+        //need to invert to make calculation compatible with dataToGrpah
+        const y1 = y.invert(
+          id.includes("valley") ? d3.select("#valleyLine").attr("y1") : d3.select("#peakLine").attr("y1")
+        );
+        const y2 = y.invert(
+          id.includes("valley") ? d3.select("#valleyLine").attr("y2") : d3.select("#peakLine").attr("y2")
+        );
+
+        const x1 = x.invert(d3.select("#peakLineY1Marker").attr("cx"));
+        const x2 = x.invert(d3.select("#peakLineY2Marker").attr("cx"));
+
+        const slope = (y2 - y1) / (x2 - x1);
+        const yintersept = y2 - slope * x2;
+        // set variables used in marker deletion calculation
+        if (id.includes("peak")) {
+          let newArr = [...peakY1];
+          newArr[wellNameToIndex[currentWell]] = y1;
+          setPeakY1(newArr);
+          newArr = [...peakY2];
+          newArr[wellNameToIndex[currentWell]] = y2;
+          setPeakY2(newArr);
+          newArr = [...peakYIntercept];
+          newArr[wellNameToIndex[currentWell]] = yintersept;
+          setPeakYIntercept(newArr);
+          //assign array
+          newArr = [...peakSlope];
+          newArr[wellNameToIndex[currentWell]] = slope;
+          setPeakSlope(newArr);
         } else {
-          pvCopy[currentWell].maxValleys = y.invert(yPosition);
-          setPeakValleyWindows({ ...pvCopy });
+          let newArr = [...valleyY1];
+          newArr[wellNameToIndex[currentWell]] = y1;
+          setValleyY1(newArr);
+          newArr = [...valleyY2];
+          newArr[wellNameToIndex[currentWell]] = y2;
+          setValleyY2(newArr);
+          newArr = [...valleyYIntercept];
+          newArr[wellNameToIndex[currentWell]] = yintersept;
+          setValleyYIntercept(newArr);
+          newArr = [...valleySlope];
+          newArr[wellNameToIndex[currentWell]] = slope;
+          setValleySlope(newArr);
         }
         // descrease stroke width when unselected and dropped
         d3.select(this).attr("stroke-width", 2);
       });
+    const moveLineUpDown = d3
+      .drag()
+      .on("start", function (d) {
+        const id = d3.select(this).attr("id");
+        // close context menu if it's open
+        contextMenu.style("display", "none");
+        // increase stroke width when selected and dragging
+        d3.select(this).attr("stroke-width", 5);
+        //set starting y position
+        const yPosition = id.includes("valley")
+          ? Math.max(d.y, y(yMax + yRange))
+          : Math.min(d.y, y(yMin - yRange));
+        d3.select(this).attr("startingY", yPosition);
+      })
+      .on("drag", function (d) {
+        const id = d3.select(this).attr("id");
+
+        const yPosition = id.includes("valley")
```

Comment:
This same code is written a few lines above, we should make a function to do this

Reply:
They are used for two different things. The one in the drag start event sets an initial y position to compare the dragging distance. The second one keeps getting the new position of where it is dragged to. Then they are compared to get the change in y